test/unit/test_annotated_code.c

```diff
@@ -0,0 +1,298 @@
+	annotation = makeRCodeAnnotation(23, 35, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_FUNCTION_NAME);
+	r_vector_push (&test_annotations, &annotation); //4
+	annotation = makeRCodeAnnotation(36, 51, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_CONSTANT_VARIABLE);
+	r_vector_push (&test_annotations, &annotation); //5
+	annotation = makeRCodeAnnotation(23, 52, R_CODE_ANNOTATION_TYPE_OFFSET, 4440, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation); //6
+	annotation = makeRCodeAnnotation(58, 64, R_CODE_ANNOTATION_TYPE_OFFSET, 4447, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation); //7
+	annotation = makeRCodeAnnotation(58, 64, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation); //8
+	annotation = makeRCodeAnnotation(58, 64, R_CODE_ANNOTATION_TYPE_OFFSET, 4447, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&test_annotations, &annotation); //9	
+	
+	return test_annotations;
+}
+
+
+static RAnnotatedCode *getHelloWorld(){
+	char *test_string = malloc(71 * sizeof(char));
+	strcpy(test_string, "\nvoid main(void)\n{\n    sym.imp.puts(\"Hello, World!\");\n    return;\n}\n");
+	RAnnotatedCode *code = r_annotated_code_new(test_string);
+
+	RVector/*<RCodeAnnotation>*/ test_annotations;
+	test_annotations = getSomeAnnotationsHelloWorld();
+	RCodeAnnotation *annotation;
+	r_vector_foreach (&test_annotations, annotation) {
+		r_annotated_code_add_annotation(code, annotation);
+	}
+	return code;
+}
+
+
+static bool test_r_annotated_code_new(){
+
+	//Testing RAnnoatedCode->code
+	char *test_string = malloc(14 * sizeof(char));
+	strcpy(test_string, "How are you?");
+	RAnnotatedCode *code = r_annotated_code_new(test_string);
+	mu_assert_streq(code->code, test_string, "Code in RAnnotatedCode is not set as expected");
+	
+	// Testing RAnnoatedCode->annotations
+	mu_assert_eq(code->annotations.elem_size, sizeof(RCodeAnnotation), "Code Annotations are initialized is not properly");
+	
+	mu_end;
+}
+
+static bool test_r_annotated_code_free(){
+
+	char *test_string = malloc(14 * sizeof(char));
+	strcpy(test_string, "How are you?");
+	RAnnotatedCode *code = r_annotated_code_new(test_string);
+
+	RCodeAnnotation test_annotation1, test_annotation2;
+	test_annotation1 = makeRCodeAnnotation(1, 2, R_CODE_ANNOTATION_TYPE_OFFSET, 123, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&code->annotations, &test_annotation1);
+	test_annotation2 = makeRCodeAnnotation(1, 5, R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT, 123, R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD);
+	r_vector_push (&code->annotations, &test_annotation2);
+	
+	// This test is only for run errors
+	r_annotated_code_free(code);
+
+	mu_end;
+}	
+
+static bool test_equal(RCodeAnnotation *first, RCodeAnnotation *second){ // First - Got, Second - Expected
+	mu_assert_eq(first->start, second->start, "start of annotations doesn't match");
```

Comment:
```suggestion
	mu_assert_eq (first->start, second->start, "start of annotations doesn't match");
```
This applies for all function/macro calls (but not their definitions).